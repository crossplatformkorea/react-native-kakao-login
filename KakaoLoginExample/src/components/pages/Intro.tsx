import {
  KakaoOAuthToken,
  KakaoProfile,
  getProfile as getKakaoProfile,
  login,
  logout,
  unlink,
} from '@react-native-seoul/kakao-login';
import React, {useState} from 'react';

import Button from '../uis/Button';
import {IC_MASK} from '../../utils/Icons';
import ResultView from '../uis/IntroTemp';
import {View} from 'react-native';
import styled from 'styled-components/native';
import {withScreen} from '../../utils/wrapper';

const Container = styled.View`
  flex: 1;
  align-self: stretch;
  overflow: scroll;
  background-color: ${({theme}) => theme.background};

  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  overflow: hidden;
`;

const ButtonWrapper = styled.View`
  position: absolute;
  flex-direction: column;
  bottom: 40px;
  width: 85%;
  align-self: center;
`;

function Intro(): React.ReactElement {
  const [result, setResult] = useState<string>('');

  const signInWithKakao = async (): Promise<void> => {
    const token: KakaoOAuthToken = await login();

    setResult(JSON.stringify(token));
  };

  const signOutWithKakao = async (): Promise<void> => {
    const message = await logout();

    setResult(message);
  };

  const getProfile = async (): Promise<void> => {
    const profile: KakaoProfile = await getKakaoProfile();

    setResult(JSON.stringify(profile));
  };

  const unlinkKakao = async (): Promise<void> => {
    const message = await unlink();

    setResult(message);
  };

  return (
    <Container>
      <ResultView result={result} />
      <ButtonWrapper>
        <Button
          testID="btn-login"
          style={{
            backgroundColor: '#FEE500',
            borderRadius: 40,
            borderWidth: 1,
          }}
          textStyle={{
            color: 'black',
            fontSize: 16,
          }}
          imgLeftSrc={IC_MASK}
          onPress={() => signInWithKakao()}
          text={'카카오 로그인'}
        />
        <View style={{marginTop: 12}} />
        <Button
          testID="btn-login"
          style={{
            backgroundColor: '#FEE500',
            borderRadius: 40,
            borderWidth: 1,
          }}
          textStyle={{
            color: 'black',
            fontSize: 16,
          }}
          imgLeftSrc={IC_MASK}
          onPress={() => getProfile()}
          text={'프로필 조회'}
        />
        <View style={{marginTop: 12}} />
        <Button
          testID="btn-login"
          style={{
            backgroundColor: '#FEE500',
            borderRadius: 40,
            borderWidth: 1,
          }}
          textStyle={{
            color: 'black',
            fontSize: 16,
          }}
          imgLeftSrc={IC_MASK}
          onPress={() => unlinkKakao()}
          text={'링크 해제'}
        />
        <View style={{marginTop: 12}} />
        <Button
          testID="btn-login"
          style={{
            backgroundColor: '#FEE500',
            borderRadius: 40,
            borderWidth: 1,
          }}
          textStyle={{
            color: 'black',
            fontSize: 16,
          }}
          imgLeftSrc={IC_MASK}
          onPress={() => signOutWithKakao()}
          text={'카카오 로그아웃'}
        />
        <View style={{marginTop: 40}} />
      </ButtonWrapper>
    </Container>
  );
}

export default withScreen(Intro);
