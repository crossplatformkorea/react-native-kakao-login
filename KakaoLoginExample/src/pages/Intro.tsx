import { Pressable, StyleSheet, Text, View } from 'react-native';
import React, { useState } from 'react';
import { login, logout, getProfile as getKakaoProfile, unlink } from '@react-native-seoul/kakao-login';
import ResultView from './IntroView';

const Intro = () => {
  const [result, setResult] = useState<string>('');

  const signInWithKakao = async (): Promise<void> => {
    try {
      const token = await login();
      setResult(JSON.stringify(token));
    } catch (err) {
      // eslint-disable-next-line no-console
      console.error('login err', err);
    }
  };

  const signOutWithKakao = async (): Promise<void> => {
    try {
      const message = await logout();

      setResult(message);
    } catch (err) {
      // eslint-disable-next-line no-console
      console.error('signOut error', err);
    }
  };

  const getProfile = async (): Promise<void> => {
    try {
      const profile = await getKakaoProfile();

      setResult(JSON.stringify(profile));
    } catch (err) {
      // eslint-disable-next-line no-console
      console.error('signOut error', err);
    }
  };

  const unlinkKakao = async (): Promise<void> => {
    try {
      const message = await unlink();

      setResult(message);
    } catch (err) {
      // eslint-disable-next-line no-console
      console.error('signOut error', err);
    }
  };

  return (
    <View style={styles.container}>
      <ResultView result={result} />
      <Pressable
        style={styles.button}
        onPress={() => {
          signInWithKakao();
        }}
      >
        <Text style={styles.text}>
          카카오 로그인
        </Text>
      </Pressable>
      <Pressable
        style={styles.button}
        onPress={() => getProfile()}
      >
        <Text style={styles.text}>
          프로필 조회
        </Text>
      </Pressable>
      <Pressable
        style={styles.button}
        onPress={() => unlinkKakao()}
      >
        <Text style={styles.text}>
          링크 해제
        </Text>
      </Pressable>
      <Pressable
        style={styles.button}
        onPress={() => signOutWithKakao()}
      >
        <Text style={styles.text}>
          카카오 로그아웃
        </Text>
      </Pressable>
    </View>
  );
};

export default Intro;

const styles = StyleSheet.create({
  container: {
    height: "100%",
    justifyContent: "flex-end",
    alignItems: 'center',
    paddingBottom: 100
  },
  button: {
    backgroundColor: '#FEE500',
    borderRadius: 40,
    borderWidth: 1,
    width: 250,
    height: 40,
    paddingHorizontal: 20,
    paddingVertical: 10,
    marginTop: 10
  },
  text: {
    textAlign: "center"
  }
});