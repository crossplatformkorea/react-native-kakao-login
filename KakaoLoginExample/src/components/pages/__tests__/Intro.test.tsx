import React, {ReactElement} from 'react';
import {
  RenderAPI,
  act,
  cleanup,
  fireEvent,
  render,
} from '@testing-library/react-native';
import {createTestElement, createTestProps} from '../../../../test/testUtils';

import Intro from '../Intro';
import {ThemeType} from 'dooboo-ui';

let props: any;
let component: ReactElement;
let testingLib: RenderAPI;

describe('[Intro] screen rendering test', () => {
  beforeEach(() => {
    props = createTestProps();
    component = createTestElement(<Intro {...props} />);
    testingLib = render(component);
  });

  afterEach(cleanup);

  it('should render outer component and snapshot matches', () => {
    testingLib = render(component);

    const baseElement = testingLib.toJSON();

    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });

  it('should render [Dark] theme', () => {
    component = createTestElement(<Intro {...props} />, ThemeType.DARK);
    testingLib = render(component);

    const baseElement = testingLib.toJSON();

    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });

  it('should render [isLoading] status', () => {
    props = createTestProps({
      isLoading: true,
    });

    component = createTestElement(<Intro {...props} />, ThemeType.DARK);
    testingLib = render(component);

    const baseElement = testingLib.toJSON();

    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });

  it('should render [isDisabled] status', () => {
    props = createTestProps({
      isDisabled: true,
    });

    component = createTestElement(<Intro {...props} />, ThemeType.DARK);
    testingLib = render(component);

    const baseElement = testingLib.toJSON();

    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });
});
