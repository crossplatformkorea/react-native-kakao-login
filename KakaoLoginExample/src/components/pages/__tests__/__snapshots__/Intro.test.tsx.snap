// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Intro] screen rendering test should render [Dark] theme 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#232323",
          "flexBasis": 0,
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "flex-start",
          "overflow": "hidden",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "height": "100%",
            "justifyContent": "flex-start",
            "paddingBottom": 24,
            "paddingLeft": 24,
            "paddingRight": 24,
            "paddingTop": 24,
            "width": "100%",
          },
        ]
      }
    >
      <RCTScrollView>
        <View>
          <Text
            style={
              Array [
                Object {
                  "color": "#D3D8E8",
                  "fontSize": 18,
                  "lineHeight": 27,
                },
              ]
            }
          >
            
          </Text>
          <View
            style={
              Object {
                "height": 400,
              }
            }
          />
        </View>
      </RCTScrollView>
    </View>
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "bottom": 40,
            "flexDirection": "column",
            "position": "absolute",
            "width": "85%",
          },
        ]
      }
    >
      <View
        activeOpacity={0.7}
        onPress={[Function]}
        testID="btn-login"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "#8A96DC",
                "borderColor": "#8A96DC",
                "borderRadius": 4,
                "borderWidth": 2,
                "height": 52,
                "justifyContent": "center",
                "width": 320,
              },
              Object {
                "backgroundColor": "#FEE500",
                "borderRadius": 40,
                "borderWidth": 1,
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "canInstrument": true,
                "createTransformer": [Function],
                "getCacheKey": [Function],
                "process": [Function],
              }
            }
            style={
              Array [
                Object {
                  "height": 24,
                  "left": 16,
                  "position": "absolute",
                  "width": 24,
                },
              ]
            }
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontSize": 14,
                },
                Object {
                  "color": "black",
                  "fontSize": 16,
                },
              ]
            }
          >
            카카오 로그인
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 12,
          }
        }
      />
      <View
        activeOpacity={0.7}
        onPress={[Function]}
        testID="btn-login"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "#8A96DC",
                "borderColor": "#8A96DC",
                "borderRadius": 4,
                "borderWidth": 2,
                "height": 52,
                "justifyContent": "center",
                "width": 320,
              },
              Object {
                "backgroundColor": "#FEE500",
                "borderRadius": 40,
                "borderWidth": 1,
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "canInstrument": true,
                "createTransformer": [Function],
                "getCacheKey": [Function],
                "process": [Function],
              }
            }
            style={
              Array [
                Object {
                  "height": 24,
                  "left": 16,
                  "position": "absolute",
                  "width": 24,
                },
              ]
            }
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontSize": 14,
                },
                Object {
                  "color": "black",
                  "fontSize": 16,
                },
              ]
            }
          >
            프로필 조회
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 12,
          }
        }
      />
      <View
        activeOpacity={0.7}
        onPress={[Function]}
        testID="btn-login"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "#8A96DC",
                "borderColor": "#8A96DC",
                "borderRadius": 4,
                "borderWidth": 2,
                "height": 52,
                "justifyContent": "center",
                "width": 320,
              },
              Object {
                "backgroundColor": "#FEE500",
                "borderRadius": 40,
                "borderWidth": 1,
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "canInstrument": true,
                "createTransformer": [Function],
                "getCacheKey": [Function],
                "process": [Function],
              }
            }
            style={
              Array [
                Object {
                  "height": 24,
                  "left": 16,
                  "position": "absolute",
                  "width": 24,
                },
              ]
            }
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontSize": 14,
                },
                Object {
                  "color": "black",
                  "fontSize": 16,
                },
              ]
            }
          >
            링크 해제
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 12,
          }
        }
      />
      <View
        activeOpacity={0.7}
        onPress={[Function]}
        testID="btn-login"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "#8A96DC",
                "borderColor": "#8A96DC",
                "borderRadius": 4,
                "borderWidth": 2,
                "height": 52,
                "justifyContent": "center",
                "width": 320,
              },
              Object {
                "backgroundColor": "#FEE500",
                "borderRadius": 40,
                "borderWidth": 1,
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "canInstrument": true,
                "createTransformer": [Function],
                "getCacheKey": [Function],
                "process": [Function],
              }
            }
            style={
              Array [
                Object {
                  "height": 24,
                  "left": 16,
                  "position": "absolute",
                  "width": 24,
                },
              ]
            }
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontSize": 14,
                },
                Object {
                  "color": "black",
                  "fontSize": 16,
                },
              ]
            }
          >
            카카오 로그아웃
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 40,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`[Intro] screen rendering test should render [isDisabled] status 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#232323",
          "flexBasis": 0,
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "flex-start",
          "overflow": "hidden",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "height": "100%",
            "justifyContent": "flex-start",
            "paddingBottom": 24,
            "paddingLeft": 24,
            "paddingRight": 24,
            "paddingTop": 24,
            "width": "100%",
          },
        ]
      }
    >
      <RCTScrollView>
        <View>
          <Text
            style={
              Array [
                Object {
                  "color": "#D3D8E8",
                  "fontSize": 18,
                  "lineHeight": 27,
                },
              ]
            }
          >
            
          </Text>
          <View
            style={
              Object {
                "height": 400,
              }
            }
          />
        </View>
      </RCTScrollView>
    </View>
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "bottom": 40,
            "flexDirection": "column",
            "position": "absolute",
            "width": "85%",
          },
        ]
      }
    >
      <View
        activeOpacity={0.7}
        onPress={[Function]}
        testID="btn-login"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "#8A96DC",
                "borderColor": "#8A96DC",
                "borderRadius": 4,
                "borderWidth": 2,
                "height": 52,
                "justifyContent": "center",
                "width": 320,
              },
              Object {
                "backgroundColor": "#FEE500",
                "borderRadius": 40,
                "borderWidth": 1,
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "canInstrument": true,
                "createTransformer": [Function],
                "getCacheKey": [Function],
                "process": [Function],
              }
            }
            style={
              Array [
                Object {
                  "height": 24,
                  "left": 16,
                  "position": "absolute",
                  "width": 24,
                },
              ]
            }
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontSize": 14,
                },
                Object {
                  "color": "black",
                  "fontSize": 16,
                },
              ]
            }
          >
            카카오 로그인
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 12,
          }
        }
      />
      <View
        activeOpacity={0.7}
        onPress={[Function]}
        testID="btn-login"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "#8A96DC",
                "borderColor": "#8A96DC",
                "borderRadius": 4,
                "borderWidth": 2,
                "height": 52,
                "justifyContent": "center",
                "width": 320,
              },
              Object {
                "backgroundColor": "#FEE500",
                "borderRadius": 40,
                "borderWidth": 1,
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "canInstrument": true,
                "createTransformer": [Function],
                "getCacheKey": [Function],
                "process": [Function],
              }
            }
            style={
              Array [
                Object {
                  "height": 24,
                  "left": 16,
                  "position": "absolute",
                  "width": 24,
                },
              ]
            }
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontSize": 14,
                },
                Object {
                  "color": "black",
                  "fontSize": 16,
                },
              ]
            }
          >
            프로필 조회
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 12,
          }
        }
      />
      <View
        activeOpacity={0.7}
        onPress={[Function]}
        testID="btn-login"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "#8A96DC",
                "borderColor": "#8A96DC",
                "borderRadius": 4,
                "borderWidth": 2,
                "height": 52,
                "justifyContent": "center",
                "width": 320,
              },
              Object {
                "backgroundColor": "#FEE500",
                "borderRadius": 40,
                "borderWidth": 1,
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "canInstrument": true,
                "createTransformer": [Function],
                "getCacheKey": [Function],
                "process": [Function],
              }
            }
            style={
              Array [
                Object {
                  "height": 24,
                  "left": 16,
                  "position": "absolute",
                  "width": 24,
                },
              ]
            }
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontSize": 14,
                },
                Object {
                  "color": "black",
                  "fontSize": 16,
                },
              ]
            }
          >
            링크 해제
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 12,
          }
        }
      />
      <View
        activeOpacity={0.7}
        onPress={[Function]}
        testID="btn-login"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "#8A96DC",
                "borderColor": "#8A96DC",
                "borderRadius": 4,
                "borderWidth": 2,
                "height": 52,
                "justifyContent": "center",
                "width": 320,
              },
              Object {
                "backgroundColor": "#FEE500",
                "borderRadius": 40,
                "borderWidth": 1,
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "canInstrument": true,
                "createTransformer": [Function],
                "getCacheKey": [Function],
                "process": [Function],
              }
            }
            style={
              Array [
                Object {
                  "height": 24,
                  "left": 16,
                  "position": "absolute",
                  "width": 24,
                },
              ]
            }
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontSize": 14,
                },
                Object {
                  "color": "black",
                  "fontSize": 16,
                },
              ]
            }
          >
            카카오 로그아웃
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 40,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`[Intro] screen rendering test should render [isLoading] status 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#232323",
          "flexBasis": 0,
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "flex-start",
          "overflow": "hidden",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "height": "100%",
            "justifyContent": "flex-start",
            "paddingBottom": 24,
            "paddingLeft": 24,
            "paddingRight": 24,
            "paddingTop": 24,
            "width": "100%",
          },
        ]
      }
    >
      <RCTScrollView>
        <View>
          <Text
            style={
              Array [
                Object {
                  "color": "#D3D8E8",
                  "fontSize": 18,
                  "lineHeight": 27,
                },
              ]
            }
          >
            
          </Text>
          <View
            style={
              Object {
                "height": 400,
              }
            }
          />
        </View>
      </RCTScrollView>
    </View>
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "bottom": 40,
            "flexDirection": "column",
            "position": "absolute",
            "width": "85%",
          },
        ]
      }
    >
      <View
        activeOpacity={0.7}
        onPress={[Function]}
        testID="btn-login"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "#8A96DC",
                "borderColor": "#8A96DC",
                "borderRadius": 4,
                "borderWidth": 2,
                "height": 52,
                "justifyContent": "center",
                "width": 320,
              },
              Object {
                "backgroundColor": "#FEE500",
                "borderRadius": 40,
                "borderWidth": 1,
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "canInstrument": true,
                "createTransformer": [Function],
                "getCacheKey": [Function],
                "process": [Function],
              }
            }
            style={
              Array [
                Object {
                  "height": 24,
                  "left": 16,
                  "position": "absolute",
                  "width": 24,
                },
              ]
            }
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontSize": 14,
                },
                Object {
                  "color": "black",
                  "fontSize": 16,
                },
              ]
            }
          >
            카카오 로그인
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 12,
          }
        }
      />
      <View
        activeOpacity={0.7}
        onPress={[Function]}
        testID="btn-login"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "#8A96DC",
                "borderColor": "#8A96DC",
                "borderRadius": 4,
                "borderWidth": 2,
                "height": 52,
                "justifyContent": "center",
                "width": 320,
              },
              Object {
                "backgroundColor": "#FEE500",
                "borderRadius": 40,
                "borderWidth": 1,
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "canInstrument": true,
                "createTransformer": [Function],
                "getCacheKey": [Function],
                "process": [Function],
              }
            }
            style={
              Array [
                Object {
                  "height": 24,
                  "left": 16,
                  "position": "absolute",
                  "width": 24,
                },
              ]
            }
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontSize": 14,
                },
                Object {
                  "color": "black",
                  "fontSize": 16,
                },
              ]
            }
          >
            프로필 조회
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 12,
          }
        }
      />
      <View
        activeOpacity={0.7}
        onPress={[Function]}
        testID="btn-login"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "#8A96DC",
                "borderColor": "#8A96DC",
                "borderRadius": 4,
                "borderWidth": 2,
                "height": 52,
                "justifyContent": "center",
                "width": 320,
              },
              Object {
                "backgroundColor": "#FEE500",
                "borderRadius": 40,
                "borderWidth": 1,
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "canInstrument": true,
                "createTransformer": [Function],
                "getCacheKey": [Function],
                "process": [Function],
              }
            }
            style={
              Array [
                Object {
                  "height": 24,
                  "left": 16,
                  "position": "absolute",
                  "width": 24,
                },
              ]
            }
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontSize": 14,
                },
                Object {
                  "color": "black",
                  "fontSize": 16,
                },
              ]
            }
          >
            링크 해제
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 12,
          }
        }
      />
      <View
        activeOpacity={0.7}
        onPress={[Function]}
        testID="btn-login"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "#8A96DC",
                "borderColor": "#8A96DC",
                "borderRadius": 4,
                "borderWidth": 2,
                "height": 52,
                "justifyContent": "center",
                "width": 320,
              },
              Object {
                "backgroundColor": "#FEE500",
                "borderRadius": 40,
                "borderWidth": 1,
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "canInstrument": true,
                "createTransformer": [Function],
                "getCacheKey": [Function],
                "process": [Function],
              }
            }
            style={
              Array [
                Object {
                  "height": 24,
                  "left": 16,
                  "position": "absolute",
                  "width": 24,
                },
              ]
            }
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontSize": 14,
                },
                Object {
                  "color": "black",
                  "fontSize": 16,
                },
              ]
            }
          >
            카카오 로그아웃
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 40,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`[Intro] screen rendering test should render outer component and snapshot matches 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#FFFFFF",
          "flexBasis": 0,
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "flex-start",
          "overflow": "hidden",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "height": "100%",
            "justifyContent": "flex-start",
            "paddingBottom": 24,
            "paddingLeft": 24,
            "paddingRight": 24,
            "paddingTop": 24,
            "width": "100%",
          },
        ]
      }
    >
      <RCTScrollView>
        <View>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontSize": 18,
                  "lineHeight": 27,
                },
              ]
            }
          >
            
          </Text>
          <View
            style={
              Object {
                "height": 400,
              }
            }
          />
        </View>
      </RCTScrollView>
    </View>
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "bottom": 40,
            "flexDirection": "column",
            "position": "absolute",
            "width": "85%",
          },
        ]
      }
    >
      <View
        activeOpacity={0.7}
        onPress={[Function]}
        testID="btn-login"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "#393D7A",
                "borderColor": "#393D7A",
                "borderRadius": 4,
                "borderWidth": 2,
                "height": 52,
                "justifyContent": "center",
                "width": 320,
              },
              Object {
                "backgroundColor": "#FEE500",
                "borderRadius": 40,
                "borderWidth": 1,
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "canInstrument": true,
                "createTransformer": [Function],
                "getCacheKey": [Function],
                "process": [Function],
              }
            }
            style={
              Array [
                Object {
                  "height": 24,
                  "left": 16,
                  "position": "absolute",
                  "width": 24,
                },
              ]
            }
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#D3D8E8",
                  "fontSize": 14,
                },
                Object {
                  "color": "black",
                  "fontSize": 16,
                },
              ]
            }
          >
            카카오 로그인
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 12,
          }
        }
      />
      <View
        activeOpacity={0.7}
        onPress={[Function]}
        testID="btn-login"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "#393D7A",
                "borderColor": "#393D7A",
                "borderRadius": 4,
                "borderWidth": 2,
                "height": 52,
                "justifyContent": "center",
                "width": 320,
              },
              Object {
                "backgroundColor": "#FEE500",
                "borderRadius": 40,
                "borderWidth": 1,
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "canInstrument": true,
                "createTransformer": [Function],
                "getCacheKey": [Function],
                "process": [Function],
              }
            }
            style={
              Array [
                Object {
                  "height": 24,
                  "left": 16,
                  "position": "absolute",
                  "width": 24,
                },
              ]
            }
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#D3D8E8",
                  "fontSize": 14,
                },
                Object {
                  "color": "black",
                  "fontSize": 16,
                },
              ]
            }
          >
            프로필 조회
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 12,
          }
        }
      />
      <View
        activeOpacity={0.7}
        onPress={[Function]}
        testID="btn-login"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "#393D7A",
                "borderColor": "#393D7A",
                "borderRadius": 4,
                "borderWidth": 2,
                "height": 52,
                "justifyContent": "center",
                "width": 320,
              },
              Object {
                "backgroundColor": "#FEE500",
                "borderRadius": 40,
                "borderWidth": 1,
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "canInstrument": true,
                "createTransformer": [Function],
                "getCacheKey": [Function],
                "process": [Function],
              }
            }
            style={
              Array [
                Object {
                  "height": 24,
                  "left": 16,
                  "position": "absolute",
                  "width": 24,
                },
              ]
            }
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#D3D8E8",
                  "fontSize": 14,
                },
                Object {
                  "color": "black",
                  "fontSize": 16,
                },
              ]
            }
          >
            링크 해제
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 12,
          }
        }
      />
      <View
        activeOpacity={0.7}
        onPress={[Function]}
        testID="btn-login"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "#393D7A",
                "borderColor": "#393D7A",
                "borderRadius": 4,
                "borderWidth": 2,
                "height": 52,
                "justifyContent": "center",
                "width": 320,
              },
              Object {
                "backgroundColor": "#FEE500",
                "borderRadius": 40,
                "borderWidth": 1,
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "canInstrument": true,
                "createTransformer": [Function],
                "getCacheKey": [Function],
                "process": [Function],
              }
            }
            style={
              Array [
                Object {
                  "height": 24,
                  "left": 16,
                  "position": "absolute",
                  "width": 24,
                },
              ]
            }
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#D3D8E8",
                  "fontSize": 14,
                },
                Object {
                  "color": "black",
                  "fontSize": 16,
                },
              ]
            }
          >
            카카오 로그아웃
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 40,
          }
        }
      />
    </View>
  </View>
</View>
`;
